From 7122ad79673279b32fdcb7e5fb9830804c6f2e15 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Wed, 10 Oct 2018 15:08:48 +0800
Subject: [PATCH 4794/5242] MLK-19863 ARM64: dts: imx8qm: Fix VPU video
 playing in Domu

commit  b2a3a2a635b36880f04e6faead31ecc90e480064 from
https://source.codeaurora.org/external/imx/linux-imx.git

commit 108e99f6487("MLK-19774 VPU: Add CSR address in dts and remove some CM4 legacy code")
did not take xen domu into consideration and cause VPU probe failure in
domu.

Fix it by add reg-csr in domu dts and passthrough the reg space.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |    4 ++++
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |    1 +
 arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi  |    4 ++++
 3 files changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index 1ac3889..fcd470e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -591,6 +591,10 @@
 			<SC_R_VPU_MU_3>;
 };
 
+&vpu_decoder_csr {
+	xen,passthrough;
+};
+
 &vpu_encoder {
        iommus = <&smmu>;
        #stream-id-cells = <1>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index 07471bd..9cb25f23 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -225,6 +225,7 @@
 			str-region = <&decoder_str>;
 			reg = <0x0 0x2c000000 0x0 0x1000000>;
 			reg-names = "vpu_regs";
+			reg-csr = <0x2d080000>;
 			power-domains = <&pd_vpu_dec>;
 			status = "disabled";
 		};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
index 05acd06..e4659d2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
@@ -188,6 +188,10 @@
 	img_pdma_3_lpcg: img_pdma_3_lpcg@58530000 {
 		reg = <0x0 0x58530000 0x0 0x1000>;
 	};
+
+	vpu_decoder_csr: vpu_decoder_csr@0x2d080000 {
+		reg = <0x0 0x2d080000 0x0 0x1000>;
+	};
 };
 
 /delete-node/ &edma0;
-- 
1.7.9.5

